doc/qos/SONiC_ROCEv2_support.md

```diff
@@ -0,0 +1,488 @@
+# Feature Name
+RDMA (Remote Direct Memory Access) over Converged Ethernet (ROCEv2) support
+
+# High Level Design Document
+
+#### Rev 0.1
+# Table of Contents
+
+  * [List of Tables](#list-of-tables)
+  * [Revision](#revision)
+  * [About This Manual](#about-this-manual)
+  * [Scope](#scope)
+  * [Definition/Abbreviation](#definitionabbreviation)
+
+# List of Tables
+[Table 1: Abbreviations](#table-1-abbreviations)
+
+# Revision
+| Rev |     Date    |       Author       | Change Description                |
+|:---:|:-----------:|:------------------:|-----------------------------------|
+| 0.1 | 06/17/2021 |   Venkatesan Mahalingam         | Initial version                   |
+
+
+
+# About this Manual
+
+This document introduces the support of ROCEv2 in mgmt-framework,
+this provides the North Bound Interface (NBI) interfaces i.e REST/gNMI/KLISH for the ROCEv2 configurations.
+
+# Scope
+
+This document covers the following,
+
+1) NBI interface to initialize QoS buffers for PFC \
+2) NBI interface to create/delete buffer pool \
+3) NBI interface to create/delete buffer profile \
+4) NBI interface to map/unmap priority group to buffer profile \
+5) Unit Testcases
+
+# Definition/Abbreviation
+
+### Table 1: Abbreviations
+| **Term**                 | **Meaning**                         |
+|--------------------------|-------------------------------------|
+| RDMA                     | Remote Direct Memory Access         |
+|          ROCEv2            |  RDMA over Converged Ethernet version 2           |
+|          PFC            |  Priority Flow Control|
+
+
+# 1 Feature Overview
+
+ROCE provides the ability to transport Infiniband (IB) protocol traffic using RDMA over lossless converged Ethernet
+across storage or compute nodes.
+ROCEv2 stands for RDMA over Converged Ethernet version 2, this is an extension of ROCE protocol a.k.a routable ROCE,
+This overcomes the limitation of ROCEv1 bounded to a single broadcast domain (VLAN),
+ROCEv2 encapsulates an RDMA transport packet within the Ethernet/IPv4/UDP packet,
+ROVEv2 uses PFC to prevent buffer overflow.
+

```

Comment:
Please update this section with Management Framework capabilities that would be used in configuring the feature. Also provide info about REST/gNMI support.

Reply:
This is feature overview, requirements section is updated with mgmt-fwk requirements.